### PR TITLE
chore: Absolute Imports and Module path aliases

### PR DESCRIPTION
**Pull request by davidsonsns**
## Description

**Issue:** #250 

Adding path alias we'll be able to import resources without `../../../../ ..Infinity`

## Subtasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] Update next to the latest version, [once this feature is only available later `9.5`](https://nextjs.org/docs/advanced-features/module-path-aliases)
- [x] Enable scripts using `ts-node` to implement alias paths adding [`tsconfig-paths`](https://www.npmjs.com/package/tsconfig-paths)
- [x] Create a `tsconfig-base` which can be also used by the scripts `diff-test`, `seed` and `build:nexus`
- [x] Create alias paths on `tsconfig-base`¹
- [x] Refactor codes with `../../` design

✋  Was required to move the file `next.d.ts` to `types/next.d.ts`. Not sure whats happened, but after creating the field `baseUrl` on `tsconfig` it wasn't working as expected. I mean, typescript was importing next types. By doing this fixed this issue.

`alias paths`
```ts
{
  "@/components/*": ["components/*"],
  "@/config/*": ["config/*"],
  "@/elements/*": ["elements/*"],
  "@/generated/*": ["generated/*"],
  "@/hooks/*": ["hooks/*"],
  "@/lib/*": ["lib/*"],
  "@/styles/*": ["styles/*"],
  "@/theme": ["theme/index"],
  "@/theme/*": ["theme/*"],
  "@/utils": ["utils/index"],
  "@/utils/*": ["utils/*"]
}
```

## Screenshots
Screen Recording 2021-01-07 at 08 03 24

### CODE REVIEW

**Review by Lanny — APPROVED** [review]
Looks awesome, very nice little developer quality of life improvement! I haven't tested thoroughly but preview env looks good and build is passing so gets a 👍  from me

**Review by robin-macpherson — APPROVED** [review]
Awesome work @davidsonsns ! This one was stuck for quite a while so it's great to see this go through and definitely a developer quality of life improvement!

File: tsconfig.json [hunk 1]
```diff
@@ -14,17 +15,13 @@
...
-      }
-    ],
+    "plugins": [{ "name": "nexus/typescript-language-service" }],
     "removeComments": true,
     "resolveJsonModule": true,
     "skipLibCheck": true,
     "sourceMap": true,
     "strict": true,
-    "target": "ES6"
+    "target": "ES6",
```
Comment (Lanny) [inline tsconfig.json:24]:
```suggestion
    "target": "ES6"
```

think this was a JS linter trying to autoformat this file, but the trailing comma is technically invalid JSON

Comment (davidsonsns) [reply tsconfig.json]: Lol it's was! Thanks for fix it

**Pull request merged**